lib/exifr/tiff.rb

```diff
@@ -256,6 +256,13 @@ class << self
             EXIFR.logger.warn("Bad date/time value #{v.inspect}: #{ex}")
             nil
           end
+        elsif v =~ /^(\d{4}):(\d\d):(\d\d) (\d\d):(\d\d):(\d\d).(\d{3})$/
+          begin
+            mktime_proc.call($1.to_i, $2.to_i, $3.to_i, $4.to_i, $5.to_i, $6.to_i, $7.to_i * 1000)
+          rescue => ex
+            EXIFR.logger.warn("Bad date/time value #{v.inspect}: #{ex}")
+            nil
+          end
```

Comment:
Please do the following instead:

```
value.map do |v|
  if v =~ /^(\d{4}):(\d\d):(\d\d) (\d\d):(\d\d):(\d\d)(?:\.(\d{3}))?$/
    begin
      mktime_proc.call($1.to_i, $2.to_i, $3.to_i, $4.to_i, $5.to_i, $6.to_i, $7.to_i * 1000)
    rescue => ex
      EXIFR.logger.warn("Bad date/time value #{v.inspect}: #{ex}")
      nil
    end
  else
    v
  end
end
```

to avoid duplicating the error handling and escapes the dot before the milliseconds.

Reply:
Thank you very much your proposal! I comitted it.